helpers.go

```diff
@@ -271,50 +271,56 @@ func setForeignKey(target interface{}, name string, source interface{}) error {
 
 // extractTag generates the map from tag to metadata
 func extractTag(dataType reflect.Type) (map[string]tagInfo, []string, error) {
-	tagToInfo := map[string]tagInfo{}
-	ignoreFields := []string{}
+	numField := dataType.NumField()
+	var ignoreFields []string
+	tagToInfo := make(map[string]tagInfo)
 
-	for i := 0; i < dataType.NumField(); i++ {
+	for i := 0; i < numField; i++ {
```

Comment:
the expression is evaluated during each iteration, so it's better to define `numField` first